pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -226,6 +228,31 @@ private String getQueryResponse(String query, @Nullable SqlNode sqlNode, String
     return sendRequestToBroker(query, instanceId, traceEnabled, queryOptions, httpHeaders);
   }
 
+  // get list of common broker instances for a given set of tables
+  private List<String> getCommonInstances(List<String> tableNames) {
+    boolean firstTable = true;
+    Set<String> brokerInstanceList = new HashSet<>();
+    for (String tableName : tableNames) {
+      String rawTableName = TableNameBuilder.extractRawTableName(_pinotHelixResourceManager
+              .getActualTableName(tableName));
+      List<String> tableBrokerInstances;
+      try {
+        tableBrokerInstances = _pinotHelixResourceManager.getLiveBrokersForTable(rawTableName);
```

Comment:
I think we shouldn't convert this to a raw table name and instead pass the tableName directly. If you use a rawTableName, the `getLiveBrokersForTable` method will return an intersection of the realtime and offline brokers for hybrid tables. If a user decides to use a type-suffix for a hybrid table, we can simply honor that and return live brokers for that type.

I am not sure what the current behavior is for those tables though (would be best to be consistent with that).

Reply:
Yeah good point! I was thinking along the same lines of removing this.